www/js/lib/settingsStore.js

```diff
@@ -98,61 +98,64 @@ define([], function () {
    * @param {String} object Optional name of the object to disable or delete ('cookie', 'localStorage', 'cacheAPI')
    */
   function reset(object) {
+    //Requiring uiUtil to handle circular dependency b/w settingsStore and uiUtil
     var uiUtil = require('uiUtil');
     // If no specific object was specified, we are doing a general reset, so ask user for confirmation
-    uiUtil.systemAlert('Confirmation', 'WARNING: This will reset the app to a freshly installed state, deleting all app caches and settings!', true)
-      .then(function (confirmation) {
-        if(!object && !confirmation) return;
+    if (!object) {
+      uiUtil.systemAlert('Confirmation', 'WARNING: This will reset the app to a freshly installed state, deleting all app caches and settings!', true)
+        .then(function (confirmation) {
+          if (!confirmation) return;
+        });
+    }
```

Comment:
Here I am facing an issue in using the `systemAlert` function.  Since the function is returning a promise, the execution thread is not waiting for it to be resolved, and hence as soon as the confirm modal appears  the rest of the code below it resets the app without waiting for the confirmation. How can I resolve this since I can't even use async/await?

Reply:
@gaurav7019 Yes, it requires some refactoring so that the rest of the code is called from within the Promise. Async/await are just sugar for working with Promises, so it's good to learn how to do it without relying on them!

However, I recognized that this is beyond the scope of this PR, and if you don't feel confident refactoring, you could leave this one with the synchronous alert/confirm to be completed in a separate PR (which I'd be happy to tackle).

One way to do it would be to put the code that must wait into a function defined with `var continue = function () {  ... };`, and call that function from with the then function. Make sure that the new function is in the same scope (it neds to access variables), and you will need to define it before the systemAlert function (otherwise it will be undefined). 

Another way to do it is to put all the code that can only run on confirm into the then function.

Reply:
Yes @Jaifroid, thanks for the info. I would like to try awaiting the code without async/await. I already had some idea on this, so I'm not completely new to this. I'll let you know otherwise (won't take long).